src/components/ChatEntry.js

```diff
@@ -1,22 +1,45 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
 
 const ChatEntry = (props) => {
+  const handleHeartChange = () => {
+    
+    const updatedChat = {
+      id: props.id,
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !props.liked
+
+    };
```

Comment:
So remember you didn't need to construct any of this, you really just needed to pass the `id` as a parameter on line 19. And we could have even done that on line 30 like so : `<button onClick ={() => props.onUpdateChat(props.id)} className="like">{heart}</button>`